4337-gas-metering/pimlico/pimlico.ts

```diff
@@ -0,0 +1,275 @@
+import dotenv from "dotenv";
+import { getAccountNonce } from "permissionless";
+import { UserOperation, bundlerActions } from "permissionless";
+import { pimlicoBundlerActions } from "permissionless/actions/pimlico";
+import { Hash, createClient, createPublicClient, http } from "viem";
+import { privateKeyToAccount } from "viem/accounts";
+import { goerli } from "viem/chains";
+import {
+  SAFE_ADDRESSES_MAP,
+  encodeCallData,
+  getAccountAddress,
+  getAccountInitCode,
+} from "../utils/safe";
+import { submitUserOperation, signUserOperation } from "../utils/userOps";
+import { setTimeout } from "timers/promises";
+import {
+  generateTransferCallData,
+  getERC20Decimals,
+  getERC20Balance,
+  mintERC20Token,
+} from "../utils/erc20";
+import { generateMintingCallData } from "../utils/erc721";
+
+dotenv.config();
+const paymaster = "pimlico";
+
+const privateKey = process.env.PRIVATE_KEY;
+
+const entryPointAddress = process.env
+  .PIMLICO_ENTRYPOINT_ADDRESS as `0x${string}`;
+const multiSendAddress = process.env.PIMLICO_MULTISEND_ADDRESS as `0x${string}`;
+
+const saltNonce = BigInt(process.env.PIMLICO_NONCE as string);
+
+const chain = process.env.PIMLICO_CHAIN;
+const chainID = Number(process.env.PIMLICO_CHAIN_ID);
+
+const safeVersion = process.env.SAFE_VERSION as string;
+
+const rpcURL = process.env.PIMLICO_RPC_URL;
+const apiKey = process.env.PIMLICO_API_KEY;
+
+const erc20PaymasterAddress = process.env
+  .PIMLICO_ERC20_PAYMASTER_ADDRESS as `0x${string}`;
+const usdcTokenAddress = process.env
+  .PIMLICO_USDC_TOKEN_ADDRESS as `0x${string}`;
+const erc20TokenAddress = process.env
+  .PIMLICO_ERC20_TOKEN_CONTRACT as `0x${string}`;
+const erc721TokenAddress = process.env
+  .PIMLICO_ERC721_TOKEN_CONTRACT as `0x${string}`;
+
+const argv = process.argv.slice(2);
+if (argv.length != 1) {
+  throw new Error("TX Type Argument not passed."); // account || erc20 || erc721
+}
+
+const txType: string = argv[0];
+if (txType != "account" && txType != "erc20" && txType != "erc721") {
+  throw new Error("TX Type Argument Invalid");
+}
+
+const safeAddresses = (
+  SAFE_ADDRESSES_MAP as Record<string, Record<string, any>>
+)[safeVersion];
+let chainAddresses;
+if (safeAddresses) {
+  chainAddresses = safeAddresses[chainID];
+}
+
+if (apiKey === undefined) {
+  throw new Error(
+    "Please replace the `apiKey` env variable with your Pimlico API key",
+  );
+}
+
+if (!privateKey) {
+  throw new Error(
+    "Please populate .env file with demo Private Key. Recommended to not use your personal private key.",
+  );
+}
+
+const signer = privateKeyToAccount(privateKey as Hash);
+console.log("Signer Extracted from Private Key.");
+
+let bundlerClient;
+let publicClient;
+if (chain == "goerli") {
```

Comment:
Not blocking approval, but we should likely use `sepolia` since goerli is dying.

Reply:
I'll add you to the slack channel with pimlico

Reply:
Actually you're already a part of the channel, so just go ahead and ask them.

Reply:
Moved to `mumbai` Polygon Testnet for the time being.